src/coreclr/System.Private.CoreLib/src/System/RuntimeHandles.cs

```diff
@@ -1895,68 +1965,96 @@ internal static int GetMDStreamVersion(RuntimeModule module)
 
     internal sealed unsafe partial class Signature
     {
-        #region FCalls
-        [MemberNotNull(nameof(m_arguments))]
-        [MemberNotNull(nameof(m_declaringType))]
-        [MemberNotNull(nameof(m_returnTypeORfieldType))]
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        private extern void GetSignature(
-            void* pCorSig, int cCorSig,
-            RuntimeFieldHandleInternal fieldHandle, IRuntimeMethodInfo? methodHandle, RuntimeType? declaringType);
-        #endregion
-
         #region Private Data Members
         //
         // Keep the layout in sync with SignatureNative in the VM
         //
-        internal RuntimeType[] m_arguments;
-        internal RuntimeType m_declaringType;
-        internal RuntimeType m_returnTypeORfieldType;
-        internal object? m_keepalive;
-        internal void* m_sig;
-        internal int m_managedCallingConventionAndArgIteratorFlags; // lowest byte is CallingConvention, upper 3 bytes are ArgIterator flags
-        internal int m_nSizeOfArgStack;
-        internal int m_csig;
-        internal RuntimeMethodHandleInternal m_pMethod;
+        private RuntimeType[]? _arguments;
+        private RuntimeType _declaringType;
+        private RuntimeType _returnTypeORfieldType;
+#pragma warning disable CA1823, 169
+        private object? _keepAlive;
+#pragma warning restore CA1823, 169
+        private void* _sig;
+        private int _csig;
+        private int _managedCallingConventionAndArgIteratorFlags; // lowest byte is CallingConvention, upper 3 bytes are ArgIterator flags
+#pragma warning disable CA1823, 169
+        private int _nSizeOfArgStack;
+#pragma warning restore CA1823, 169
+        private RuntimeMethodHandleInternal _pMethod;
         #endregion
 
+        [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "Signature_Init")]
+        private static partial void Init(
+            ObjectHandleOnStack _this,
+            void* pCorSig, int cCorSig,
+            RuntimeFieldHandleInternal fieldHandle,
+            RuntimeMethodHandleInternal methodHandle,
+            IntPtr typeHandle);
```

Comment:
`IntPtr typeHandle` argument is unnecessary (the type can be read from `_this` on the unmanaged side).